miden-lib/src/transaction/inputs.rs

```diff
@@ -229,13 +229,6 @@ fn add_account_to_advice_inputs(
     // extend the merkle store with account code tree
     inputs.extend_merkle_store(code.procedure_tree().inner_nodes());
 
-    // extend advice map with account proc root |-> proc index
```

Comment:
Most of this PR is actually so that we can remove these lines of code :)

Basically, previously mapping `proc root |-> proc index` was stored in the advice provider. This could have led to collisions (as explained in the lined issue) but also could be a potential attack vector.

Now, the mapping is maintained in the `TransactionHost` and provided to the VM when `PushAccountProcedureIndex` event is invoked.